pkg/agent/util/net_windows.go

```diff
@@ -524,5 +508,76 @@ func HostInterfaceExists(ifaceName string) bool {
 func SetInterfaceMTU(ifaceName string, mtu int) error {
 	cmd := fmt.Sprintf("Set-NetIPInterface -IncludeAllCompartments -InterfaceAlias \"%s\" -NlMtuBytes %d",
 		ifaceName, mtu)
-	return InvokePSCommand(cmd)
+	_, err := ps.RunCommand(cmd)
+	return err
+}
+
+func NewNetRoute(linkIndex int, dstSubnet *net.IPNet, gwAddr net.IP, metric int) error {
+	cmd := fmt.Sprintf("New-NetRoute -InterfaceIndex %v -DestinationPrefix %v -NextHop %v -RouteMetric %d -Verbose",
+		linkIndex, dstSubnet.String(), gwAddr.String(), metric)
+	_, err := ps.RunCommand(cmd)
+	return err
+}
+
+func RemoveNetRoute(linkIndex int, dstSubnet *net.IPNet, gwAddr net.IP) error {
+	cmd := fmt.Sprintf("Remove-NetRoute -InterfaceIndex %v -DestinationPrefix %v -NextHop  %v -Verbose -Confirm:$false",
+		linkIndex, dstSubnet.String(), gwAddr.String())
+	_, err := ps.RunCommand(cmd)
+	return err
+}
+
+func GetNetRoutes(linkIndex int, dstSubnet *net.IPNet) ([]Route, error) {
+	cmd := fmt.Sprintf("Get-NetRoute -InterfaceIndex %d -DestinationPrefix %s -erroraction Ignore | Format-Table -HideTableHeaders",
+		linkIndex, dstSubnet.String())
+	return getNetRoutes(cmd)
+}
+
+func GetNetRoutesAll() ([]Route, error) {
+	cmd := "Get-NetRoute -ErrorAction Ignore | Format-Table -HideTableHeaders"
+	return getNetRoutes(cmd)
+}
+
+func getNetRoutes(cmd string) ([]Route, error) {
+	routesStr, err := ps.RunCommand(cmd)
+	if err != nil {
+		return nil, err
+	}
+	routes, err := parseRoutes(routesStr)
+	if err != nil {
+		return nil, err
+	}
+	return routes, nil
+}
+
+func parseRoutes(routesStr string) ([]Route, error) {
+	scanner := bufio.NewScanner(strings.NewReader(routesStr))
+	var routes []Route
+	for scanner.Scan() {
+		items := strings.Fields(scanner.Text())
+		if len(items) < 4 {
+			continue
+		}
+
+		idx, err := strconv.Atoi(items[0])
+		if err != nil {
+			continue
+		}
+		_, dstSubnet, err := net.ParseCIDR(items[1])
+		if err != nil {
+			continue
+		}
+		gw := net.ParseIP(items[2])
+		metric, err := strconv.Atoi(items[3])
+		if err != nil {
+			continue
```

Comment:
so we silently ignore all errors?

Reply:
I think maybe it is ok since gonetsh ignores these?
https://github.com/rakelkar/gonetsh/blob/master/netroute/netroute.go#L112

Reply:
I'm not sure that's a good reference. You modified the function so that it supports returning an error, but you never actually return one... Not a big deal, but I am not sure why bother with an `error` return value then?

Reply:
To return error is a mistake with current design... I will think it over again if we need to return error.